Lesson25/Gokhan/filter.js

```diff
@@ -0,0 +1,422 @@
+	},
+	{
+		name: 'Fitness Tracker',
+		brand: 'Fitbit',
+		price: 129,
+		quantity: 14,
+	},
+];
+
+const emptyStock = productsExercise6.filter((stock) => stock.quantity <= 0);
+console.log('Exercise 6 - ', 'Empty Stocks Are: ', emptyStock);
+
+// Exercise 7:
+// Given an array of numbers, filter out the numbers that are divisible by 3.
+const numbersExercise7 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const divisibleBy3 = numbersExercise7.filter((number) => number % 3 === 0);
+console.log('Exercise 7 - ', 'Divisible Numbers With 3: ', divisibleBy3);
+
+// Exercise 8:
+// Given an array of strings, filter out the strings that have more than 5 characters.
+const wordsExercise8 = [
+	'apple',
+	'banana',
+	'car',
+	'dog',
+	'elephant',
+	'flower',
+	'guitar',
+	'house',
+	'island',
+	'jungle',
+];
+
+const moreThan5Words = wordsExercise8.filter((words) => words.length <= 5);
+console.log('Exercise 8 - ', 'More Than 5 Characters Of Words: ', moreThan5Words);
+
+// Exercise 9:
+// Given an array of objects representing students, filter out the students who have a grade lower than C.
+const studentsExercise9 = [
+	{
+		id: 1,
+		name: 'Alice',
+		age: 18,
+		grade: 'A',
+	},
+	{
+		id: 2,
+		name: 'Bob',
+		age: 19,
+		grade: 'B',
+	},
+	{
+		id: 3,
+		name: 'Charlie',
+		age: 20,
+		grade: 'B+',
+	},
+	{
+		id: 4,
+		name: 'David',
+		age: 19,
+		grade: 'A-',
+	},
+	{
+		id: 5,
+		name: 'Emma',
+		age: 18,
+		grade: 'A+',
+	},
+	{
+		id: 6,
+		name: 'Frank',
+		age: 20,
+		grade: 'B',
+	},
+	{
+		id: 7,
+		name: 'Grace',
+		age: 19,
+		grade: 'D+',
+	},
+	{
+		id: 8,
+		name: 'Hannah',
+		age: 20,
+		grade: 'A-',
+	},
+	{
+		id: 9,
+		name: 'Isaac',
+		age: 18,
+		grade: 'D',
+	},
+	{
+		id: 10,
+		name: 'Julia',
+		age: 19,
+		grade: 'B-',
+	},
+];
+
+const gradeLowerThanC = studentsExercise9.filter((students) => students.grade >= 'C');
+console.log('Exercise 9 - ', 'Grade Lower Than C Of Studebts: ', gradeLowerThanC);
+
+// Exercise 10:
+// Given an array of user objects, filter out the users who have a name starting with the letter "J".
+const usersExercise10 = [
+	{
+		id: 1,
+		name: 'John',
+		age: 25,
+		email: 'john@example.com',
+	},
+	{
+		id: 2,
+		name: 'Alice',
+		age: 32,
+		email: 'alice@example.com',
+	},
+	{
+		id: 3,
+		name: 'Bob',
+		age: 28,
+		email: 'bob@example.com',
+	},
+	{
+		id: 4,
+		name: 'Emily',
+		age: 21,
+		email: 'emily@example.com',
+	},
+	{
+		id: 5,
+		name: 'Michael',
+		age: 35,
+		email: 'michael@example.com',
+	},
+	{
+		id: 6,
+		name: 'Sara',
+		age: 29,
+		email: 'sara@example.com',
+	},
+	{
+		id: 7,
+		name: 'David',
+		age: 42,
+		email: 'david@example.com',
+	},
+	{
+		id: 8,
+		name: 'Olivia',
+		age: 31,
+		email: 'olivia@example.com',
+	},
+	{
+		id: 9,
+		name: 'Daniel',
+		age: 27,
+		email: 'daniel@example.com',
+	},
+	{
+		id: 10,
+		name: 'Sophia',
+		age: 23,
+		email: 'sophia@example.com',
+	},
+];
+
+const startingNameWithJ = usersExercise10.filter((user) => user.name.startsWith ('J'));
```

Comment:
Don't forget to convert the incoming value to the case that you are comparing to.
```suggestion
const startingNameWithJ = usersExercise10.filter((user) => user.name.toUpperCase().startsWith('J'));
```